Sources/MMIO/Documentation.docc/Core-Concepts/Register-Layouts.md

```diff
@@ -0,0 +1,97 @@
+# Defining Register Layouts
+
+Structure your hardware interfaces with ``MMIO/RegisterBlock()`` and ``MMIO/Register``.
+
+## Overview
+
+Swift MMIO uses a hierarchical, declarative approach for defining hardware interfaces, mirroring typical hardware organization: peripherals (or register blocks) contain individual registers, which comprise bit fields. This is achieved with two main macros: ``MMIO/RegisterBlock()`` and ``MMIO/Register(bitWidth:)``.
+
+This article provides a conceptual overview. For a step-by-step guide, see <doc:Defining-Registers>.
+
+## `@RegisterBlock`: Grouping Registers into Addressable Blocks
+
+The ``MMIO/RegisterBlock()`` macro is used to define a Swift `struct` that represents a hardware peripheral or a distinct, addressable block of registers within a larger peripheral. It serves as a container for:
+- Instances of ``MMIO/Register`` (representing individual hardware registers).
+- Other nested `RegisterBlock`s (representing sub-modules or grouped registers).
+- Arrays of registers or register blocks using ``MMIO/RegisterArray``.
+
+A `RegisterBlock` is initialized with a base memory address. Its members (registers or nested blocks) are defined with offsets relative to this parent address. `RegisterBlock`s can be nested to model complex hardware. For each member annotated with `@RegisterBlock(offset: ...)`, a computed property is generated, returning an initialized instance of that member with its `unsafeAddress` correctly calculated.
+
+### Example Structure
+```swift
+import MMIO
+
+// Top-level peripheral, e.g., a UART controller
+@RegisterBlock
+struct UARTPeripheral {
+    // Base address provided at instantiation. Member offsets are relative.
+
+    @RegisterBlock(offset: 0x00) // Data Register at +0x00
+    var data: Register<UARTData>
+
+    @RegisterBlock(offset: 0x04) // Status Register at +0x04
+    var status: Register<UARTStatus>
+
+    // Control registers grouped into a nested block at +0x08
+    @RegisterBlock(offset: 0x08)
+    var controlSubBlock: UARTControlBlock
+}
+
+// Nested register block for control-related registers
+@RegisterBlock
+struct UARTControlBlock {
+    // Offsets are relative to UARTControlBlock's effective address
+    // (UARTPeripheral.baseAddress + 0x08).
+
+    @RegisterBlock(offset: 0x00) // Main control at +0x00 within this sub-block
+    var mainCtrl: Register<UARTMainControl>
+
+    @RegisterBlock(offset: 0x04) // Baud rate config at +0x04 within this sub-block
+    var baudRate: Register<UARTBaudRate>
+}
+
+// Register structs (UARTData, UARTStatus, etc.) are defined
+// using @Register, as shown below.
+```
+
+## `@Register`: Defining Individual Register Bit Fields
```

Comment:
The backticks show up still as backticks in the rendered version... Is there some docc magic we're missing?